atlasdb-feign/src/main/java/com/palantir/atlasdb/http/v2/ConjureJavaRuntimeTargetFactory.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http.v2;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.config.AuxiliaryRemotingParameters;
+import com.palantir.atlasdb.config.ImmutableServerListConfig;
+import com.palantir.atlasdb.config.ServerListConfig;
+import com.palantir.atlasdb.http.AtlasDbRemotingConstants;
+import com.palantir.atlasdb.http.ImmutableInstanceAndVersion;
+import com.palantir.atlasdb.http.PollingRefreshable;
+import com.palantir.atlasdb.http.TargetFactory;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+import com.palantir.conjure.java.client.jaxrs.JaxRsClient;
+import com.palantir.conjure.java.config.ssl.TrustContext;
+import com.palantir.conjure.java.ext.refresh.Refreshable;
+import com.palantir.conjure.java.okhttp.HostMetricsRegistry;
+
+public final class ConjureJavaRuntimeTargetFactory implements TargetFactory {
+    private static final HostMetricsRegistry HOST_METRICS_REGISTRY = new HostMetricsRegistry();
+
+    public static final ConjureJavaRuntimeTargetFactory DEFAULT = new ConjureJavaRuntimeTargetFactory();
+    public static final String CLIENT_VERSION_STRING = "Conjure-Java-Runtime";
+
+    private ConjureJavaRuntimeTargetFactory() {
+        // Use the instances.
+    }
+
+    @Override
+    public <T> InstanceAndVersion<T> createProxy(Optional<TrustContext> trustContext, String uri, Class<T> type,
```

Comment:
we're not really using the parameters portion i.e. we're not passing in `shouldRetry`.

Reply:
good call! Fixed.

Reply:
Also added a comment on the TargetFactory interface explaining that `shouldRetry` is (correctly) ignored for the failover methods